crates/services/p2p/src/p2p_service.rs

```diff
@@ -155,62 +159,41 @@ pub enum FuelP2PEvent {
 
 impl<Codec: NetworkCodec> FuelP2PService<Codec> {
     pub fn new(config: Config, codec: Codec) -> Self {
-        let local_peer_id = PeerId::from(config.keypair.public());
-
         let gossipsub_data =
             GossipsubData::with_topics(GossipsubTopics::new(&config.network_name));
         let network_metadata = NetworkMetadata { gossipsub_data };
 
         // configure and build P2P Service
-        let (transport, connection_state) = build_transport(&config);
+        let (transport_function, connection_state) = build_transport_function(&config);
         let behaviour = FuelBehaviour::new(&config, codec.clone());
 
-        let total_connections = {
-            let reserved_nodes_count = u32::try_from(config.reserved_nodes.len())
-                .expect("The number of reserved nodes should be less than `u32::max`");
-            // Reserved nodes do not count against the configured peer input/output limits.
-            let total_peers = config
-                .max_peers_connected
-                .saturating_add(reserved_nodes_count);
-
-            total_peers.saturating_mul(config.max_connections_per_peer)
-        };
-
-        let max_established_incoming = {
-            if config.reserved_nodes_only_mode {
-                // If this is a guarded node,
-                // it should not receive any incoming connection requests.
-                // Rather, it will send outgoing connection requests to its reserved nodes
-                0
-            } else {
-                total_connections / 2
-            }
-        };
-
-        let connection_limits = ConnectionLimits::default()
```

Comment:
Good catch. We should probably use the connection_limit behavior:
https://github.com/libp2p/rust-libp2p/blob/master/misc/connection-limits/src/lib.rs

Since this isn't covered tests, I'll create a new issue for this that includes tests for proving desired behavior.

Reply:
https://github.com/FuelLabs/fuel-core/issues/1560